### PR TITLE
[CUMULUS-2474] Add S3ObjectStore to aws-client

### PR DESCRIPTION
**Pull request by dopeters**
**Summary:** Uses Harmony/frappe code for object store (https://github.com/nasa/harmony/blob/main/app/util/object-store.ts)

Addresses [CUMULUS-2474: Implement ObjectStore and S3ObjectStore](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2474)

## Changes

* Adds S3ObjectStore class and unit tests

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [ ] Adhoc testing
- [ ] Integration tests

### CODE REVIEW

File: packages/aws-client/src/S3ObjectStore.ts
Comment (markdboyd) [inline packages/aws-client/src/S3ObjectStore.ts:18]: There's a subtle difference between this and the Harmony code in that the Harmony code seems to support overriding the `s3` class used via the constructor. It might be nice to add that feature for the sake of testing, which would eliminate the need to use sinon

Comment (dopeters) [reply packages/aws-client/src/S3ObjectStore.ts:18]: Ah yes, I hadn't considered that. Will add.

File: packages/aws-client/src/S3ObjectStore.ts
Comment (markdboyd) [inline packages/aws-client/src/S3ObjectStore.ts:23]: Sorry, I should have been more clear in my feedback. I meant something a bit simpler like this:

```suggestion
  constructor(overrides?: object) {
    this.s3 = overrides.s3 || s3();
```

And then remove the `_getS3` method entirely. I'm wondering if adding the `@cumulus/common` dependency to this package is part of the problem (though I can't see why). Still, I don't love having to repeat that logic for setting up the S3 service that we already have in this aws-client package

Comment (dopeters) [reply packages/aws-client/src/S3ObjectStore.ts]: Just to summarize from Slack:
Removing the `@cumulus/common` package fixed the build errors
Since `s3()` handles test conditions (and for ease) I removed the `_getS3` method and the `overrides` param. While we're still staging things to s3 this way, we're no longer using sinon unnecessarily.

File: packages/aws-client/src/S3ObjectStore.ts
Comment (laurenfrederick) [inline packages/aws-client/src/S3ObjectStore.ts:56]: We can discuss this in Slack or the arch meeting, but I feel like this should not be in the AWS client package. I actually don't know where it should go though.

I think it could be fine here for now or maybe the S3ObjectStore stays here and these functions could move out. To where I don't know.

**Review by Jkovarik — CHANGES_REQUESTED** [review]
This looks good! Just a couple of non-blocking 'nits'/minor comments and the structure concern Lauren brought up needs addressing.

File: packages/aws-client/src/S3ObjectStore.ts
Comment (dopeters) [reply packages/aws-client/src/S3ObjectStore.ts:46]: Yeah after digging into this, it appears that this is a workaround for allowing custom query params in signed S3 urls. If we don't need to support that, we can instead use `this.s3.getSignedUrl` which typescript would recognize. I like the flexibility of being able to use custom params, but am not sure if it's a valid use case

**Comment by dopeters** [comment]
@Jkovarik Thanks for the thorough review! I believe I've addressed all the feedback. Let me know if there's anything else.
Alpha version of the package is published.

I also updated the audit whitelist to include "underscore" in this PR. I opened another PR with only that change in case this needs more changes before merging. (https://github.com/nasa/cumulus/pull/2216).

File: CHANGELOG.md
Comment (jennyhliu) [reply CHANGELOG.md]: Need to move it to the top of the CHANGLOG.md, it's not part of release 9.0.0

**Comment by Jkovarik** [comment]
@dopeters  I'm planning on taking a look this morning!

File: packages/aws-client/src/S3ObjectStore.ts
Comment (Jkovarik) [reply packages/aws-client/src/S3ObjectStore.ts]: I don't think this has been addressed - we still have harmony comments in the jsdoc for `defaultObjectStore`

File: packages/object-store/package.json
Comment (Jkovarik) [inline packages/object-store/package.json:3]: I think we may want to explicitly specify what's included in the published package, similar to what @cumulus/checksum is doing.    

e.g. 

```
npm notice
npm notice 📦  @cumulus/object-store@9.0.1-alpha.0
npm notice === Tarball Contents ===
npm notice 986B  README.md
npm notice 805B  dist/index.d.ts
npm notice 281B  dist/index.d.ts.map
npm notice 1.4kB dist/index.js
npm notice 526B  dist/index.js.map
npm notice 1.1kB package.json
npm notice 1.2kB src/index.ts
npm notice 895B  tests/test-index.js
npm notice 91B   tsconfig.json
npm notice === Tarball Details ===
```




Right now it just defaults to `*`, which will include src, tests, etc, which will work, but we might forget to tidy it up before it goes non-alpha.

**Review by Jkovarik — APPROVED** [review]
Great work!    Thanks for addressing my concerns, and thanks for your patience this morning on the re-review.  Just one more thing for consideration, approving pending resolution of that concern.

**Pull request merged**
